packages/ui/src/components/va-file-upload/VaFileUploadGalleryItem/VaFileUploadGalleryItem.vue

```diff
@@ -46,29 +48,38 @@
 import { defineComponent, onMounted, ref, watch, computed, PropType } from 'vue'
 
 import { colorToRgba } from '../../../services/color-config/color-functions'
-import { useFocus, useBem } from '../../../composables'
+import { useFocus, useBem, useStrictInject } from '../../../composables'
 
-import type { ConvertedFile } from '../types'
+import { VaFileUploadKey, ConvertedFile, VaFileUploadInject } from '../types'
 
 import { VaButton, VaListItem, VaListItemSection } from '../../index'
 import { VaFileUploadUndo } from '../VaFileUploadUndo'
 
+const INJECTION_ERROR_MESSAGE = 'The VaFileUploadGalleryItem component should be used in the context of VaFileUpload component'
+
 export default defineComponent({
   name: 'VaFileUploadGalleryItem',
+
   components: {
     VaFileUploadUndo,
     VaButton,
     VaListItem,
     VaListItemSection,
   },
+
   emits: ['remove'],
+
   props: {
     file: { type: Object as PropType<ConvertedFile>, default: null },
     color: { type: String, default: 'success' },
-    undo: { type: Boolean, default: false },
-    undoDuration: { type: Number, default: 3000 },
   },
+
   setup (props, { emit }) {
+    const {
+      undo,
+      disabled,
+      undoDuration,
+    } = useStrictInject<VaFileUploadInject>(VaFileUploadKey, INJECTION_ERROR_MESSAGE)
```

Comment:
`<VaFileUploadInject>` actually does nothing. You have return type defined in `VaFileUploadKey`, which TypeScript can use.